File: frontend/src/api/stats.ts
import { request } from './client'

export interface Stats {
  photos: number
  storage_bytes: number
  folders: number
  missing: number
  duplicate_photos: number
  reclaimable_bytes: number
  last_scan_at: string | null
  /** Lifetime tallies from the audit log — persist across rescans and rebuilds. */
  deleted_count: number
  space_saved_bytes: number
}

export function getStats(): Promise<Stats> {
  return request<Stats>('/api/stats')
}

File: backend/app/api/stats.py
from datetime import datetime
from typing import Annotated

from fastapi import APIRouter, Depends
from pydantic import BaseModel
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.services.stats import compute_stats

router = APIRouter(prefix="/stats")


class StatsRead(BaseModel):
    photos: int
    storage_bytes: int
    folders: int
    missing: int
    duplicate_photos: int
    reclaimable_bytes: int
    last_scan_at: datetime | None
    deleted_count: int
    space_saved_bytes: int


@router.get("")
def get_stats(db: Annotated[Session, Depends(get_db)]) -> StatsRead:
    stats = compute_stats(db)
    return StatsRead(
        photos=stats.photos,
        storage_bytes=stats.storage_bytes,
        folders=stats.folders,
        missing=stats.missing,
        duplicate_photos=stats.duplicate_photos,
        reclaimable_bytes=stats.reclaimable_bytes,
        last_scan_at=stats.last_scan_at,
        deleted_count=stats.deleted_count,
        space_saved_bytes=stats.space_saved_bytes,
    )
